test/irb/test_irb.rb

```diff
@@ -877,7 +877,7 @@ def bar
       end
 
       assert_match(/irbtest-.*\.rb:2:in (`|'Object#)foo': error \(RuntimeError\)/, output)
-      frame_traces = output.split("\n").select { |line| line.strip.match?(/from /) }.map(&:strip)
+      frame_traces = output.split("\n").filter_map { |line| line.strip if line.strip.match?(/from /) }
```

Comment:
Other part of this pull request looks good.
This part, I', not sure if this change is obviously better than before.

How about changing it to `.map(&:strip).grep(/from /)` ?

Reply:
Yes, that is much better!